Tools/autotest/arducopter.py

```diff
@@ -3949,6 +3949,50 @@ def fly_payload_place_mission(self):
         if ex is not None:
             raise ex
 
+    def weathervane_test(self):
+        # We test nose into wind code paths and yaw direction here and test side into wind
+        # yaw direction in QuadPlane tests to reduce repetition.
+        self.set_parameter("SIM_WIND_SPD", 10)
+        self.set_parameter("SIM_WIND_DIR", 100)
+        # allow weathervaning and arming from tx in guided
+        self.set_parameter("GUID_OPTIONS", 129)
+        # allow arming in auto, take off without raising the stick, and weathervaning
+        self.set_parameter("AUTO_OPTIONS", 131)
+        self.set_parameter("WVANE_ENABLE", 1)
+        self.set_parameter("WVANE_GAIN", 3)
+        self.set_parameter("WVANE_VELZ_MAX", 1)
+        self.set_parameter("WVANE_SPD_MAX", 2)
```

Comment:
Peter switched to this for set_parameters as it is faster

```suggestion
        self.set_parameters({
            "SIM_WIND_SPD": 10,
            "SIM_WIND_DIR": 100,
            "GUID_OPTIONS": 129,    # allow weathervaning and arming from tx in guided
            "AUTO_OPTIONS": 131,    # allow arming in auto, take off without raising the stick, and weathervaning
            "WVANE_ENABLE": 1,
            "WVANE_GAIN"  : 3,
            "WVANE_VELZ_MAX": 1,
            "WVANE_SPD_MAX" : 2
        })
```

Reply:
Done, Thanks